### PR TITLE
[kms/yubikey] Add decryption support to Yubikey KMS plugin

### PR DESCRIPTION
**Pull request by fuhry**
Some provisioners in `step-ca` (i.e. SCEP) require the ability to decrypt challenges using the intermediate CA key. [1] Although the Yubikey hardware supports decryption using RSA keys, support for this was missing from the Yubikey KMS plugin.

This commit implements the `CreateDecrypter` method in the Yubikey KMS plugin. Tests are included.

[1] https://github.com/smallstep/certificates/blob/361e2b2907bc94744da98ece290fdc1d851951f4/authority/authority.go#L540-L545

Signed-Off-By: Dan Fuhry <dan@fuhry.com>

### CODE REVIEW

**Comment by fuhry** [comment]
TODO: the CreateSigner and CreateDecrypter methods could be DRYed out a bit

**Comment by maraino** [comment]
Nice, I wanted to do this at some point.

File: kms/yubikey/yubikey.go [hunk 1]
```diff
@@ -219,6 +219,42 @@ func (k *YubiKey) CreateSigner(req *apiv1.CreateSignerRequest) (crypto.Signer, e
...
+		return nil, err
+	}
+
+	priv, err := k.yk.PrivateKey(slot, pub, piv.KeyAuth{
+		PIN:       pin,
+		PINPolicy: piv.PINPolicyAlways,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving private key")
+	}
```
Comment (maraino) [inline kms/yubikey/yubikey.go:249]: Although `CreateSigner` is implemented this way, in other KMSs, to create a signer, you need to `CreateKey -> CreateSigner`. We should use the same logic for `CreateDecrypter`, `CreateKey -> CreateDecrypter`.

I'll add a PR to fix `CreateSigner`

**Review by maraino — CHANGES_REQUESTED** [review]

File: kms/yubikey/yubikey.go (hunk 1 above)
Comment (maraino) [reply kms/yubikey/yubikey.go:249]: No, I though this method was generating the key, I'm wrong.

**Comment by fuhry** [comment]
Thanks for merging this!

It turns out that this PR on its own is not enough. The `*pki.keyRSA` (cast to `crypto.Decrypter`) returned from this doesn't match the cases checked in [pkcs7/decrypt.go](https://github.com/smallstep/pkcs7/blob/5660c44b3d3b8a1c4a125dee9ecfd658efb3f437/decrypt.go#L34), so [here's a PR to fix that](https://github.com/smallstep/pkcs7/pull/3).

**Pull request merged**